app/models/user.rb

```diff
@@ -10,9 +10,8 @@ class User < ApplicationRecord
 
   def self.new_with_session(params, session)
     super.tap do |user|
-      if data = session["devise.facebook_data"] && session["devise.facebook_data"]["extra"]["raw_info"]
-        user.email = data["email"] if user.email.blank?
-      end
+      data = session.dig(:'devise.facebook_data', :extra, :raw_info)
+      user.email = data['email'] if data && user.email.blank?
```

Comment:
should we check if email is present in the request?
```suggestion
      user.email = data['email'] if user.email.blank? && data && data['email']
```